### PR TITLE
Onboarding Replication fabrics, protection container and protection container mapping cmdlets using autorest

### PR DESCRIPTION
**Pull request by ju993rn4ut**
small changes in docs and examples

Onboarding replication protection container cmdlets

changed fabric input in new protection container

onboarding protection container mapping cmdlets

## Description

I have added all the get, remove, update, new replication fabrics, protection container and protection container mapping commands.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Remove-AzRecoveryServicesReplicationPolicy.ps1 [hunk 1]
```diff
@@ -1,6 +1,6 @@
 function Remove-AzRecoveryServicesReplicationPolicy
 {
-	[OutputType([Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Models.Api20230201.IJob])]
+	[OutputType([System.Boolean])]
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Remove-AzRecoveryServicesReplicationPolicy.ps1:3]: Indent left

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/ProtectionContainerMapping/New-AzRecoveryServicesReplicationProtectionContainerMapping.ps1 [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        # Use the default credentials for the proxy
+        ${ProxyUseDefaultCredentials}
+    )
+
+    process {
+        try {
+            $TargetProtectionContainerId = $RecoveryProtectionContainer.Id
+            $PolicyId = $Policy.Id
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/ProtectionContainerMapping/New-AzRecoveryServicesReplicationProtectionContainerMapping.ps1:166]: For these inputs, please add null checks

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Fabric/Get-AzRecoveryServicesReplicationFabric_GetByName.ps1 [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        [Parameter(DontShow)]
+        [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        # Use the default credentials for the proxy
+        ${ProxyUseDefaultCredentials}
+    )
+
+    process {
+        try {
+            $null = $PSBoundParameters.Remove("FriendlyName")
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Fabric/Get-AzRecoveryServicesReplicationFabric_GetByName.ps1]: Are we just doing it by friendly name and not fabric name  ?

**Review by vidyadharijami — CHANGES_REQUESTED** [review]

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/ProtectionContainerMapping/New-AzRecoveryServicesReplicationProtectionContainerMapping.ps1 (hunk 2 above)
Comment (ju993rn4ut) [reply src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/ProtectionContainerMapping/New-AzRecoveryServicesReplicationProtectionContainerMapping.ps1:166]: But these parameters are already mandatory, so adding null checks are useless ig

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Fabric/Get-AzRecoveryServicesReplicationFabric_GetByName.ps1 (hunk 3 above)
Comment (ju993rn4ut) [reply src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Fabric/Get-AzRecoveryServicesReplicationFabric_GetByName.ps1]: We have 2 commands one doing only with fabric name and another doing only with friendly name.

Comment (vidyadharijami) [reply src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Fabric/Get-AzRecoveryServicesReplicationFabric_GetByName.ps1]: No, let's not have two commands, we should have one single command with different parametersets like we have in existing module.

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/ProtectionContainerMapping/New-AzRecoveryServicesReplicationProtectionContainerMapping.ps1 (hunk 2 above)
Comment (vidyadharijami) [reply src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/ProtectionContainerMapping/New-AzRecoveryServicesReplicationProtectionContainerMapping.ps1:166]: As discussed, Check for null and empty

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/ProtectionContainer/Get-AzRecoveryServicesReplicationProtectionContainer.ps1 [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+<#
+.Synopsis
+Gets the details of a protection container.
+.Description
+Gets the details of a protection container.
+.Example
+{{ Add code here }}
+.Example
+{{ Add code here }}
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/ProtectionContainer/Get-AzRecoveryServicesReplicationProtectionContainer.ps1]: Remove this

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/ProtectionContainer/Remove-AzRecoveryServicesReplicationProtectionContainer.ps1 [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        [Parameter(DontShow)]
+        [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        # Use the default credentials for the proxy
+        ${ProxyUseDefaultCredentials}
+    )
+
+    process {
+        try {
+            $protectionContainerString = $ProtectionContainer.id.Split("/")
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/ProtectionContainer/Remove-AzRecoveryServicesReplicationProtectionContainer.ps1:126]: Please add validation on ARM format

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Fabric/New-AzRecoveryServicesReplicationFabric.ps1 [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+        [Parameter()]
+        [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Category('Path')]
+        [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Runtime.DefaultInfo(Script='(Get-AzContext).Subscription.Id')]
+        [System.String]
+        # The subscription Id.
+        ${SubscriptionId},
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Fabric/New-AzRecoveryServicesReplicationFabric.ps1]: - ValidateNotNull should be enough for objects

**Review by vidyadharijami — CHANGES_REQUESTED** [review]

**Review by vidyadharijami — APPROVED** [review]

**Pull request merged**